src/styles/calendar.scss

```diff
@@ -0,0 +1,46 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  padding: 10px;
+  max-width: $day-size * 7 + 6 * $gap;
+
+  @each $day, $size in $days-of-week {
+    &--start-day-#{$day} {
+      .calendar__day:first-child {
+        margin-left: $size * $start-day-size;
+      }
+    }
+  }
+
+  @each $length in $month-length {
+    &--month-length-#{$length} {
+      .calendar__day:nth-child(n + #{$length + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  &__day {
+    position: relative;
+    width: $day-size;
+    height: $day-size;
+    box-sizing: border-box;
+    background-color: $main-bg-color;
+    border: 1px solid $main-black-color;
+    transition: all 0.5s ease;
+
+    @for $days from 1 through 31 {
+      &:nth-child(#{$days})::before {
+        content: "#{$days}";
+        margin: auto;
```

Comment:
- Specify to which properties should transition apply, and actually now it doesn't work
- `margin: auto;` is the same for all children of the day, so you can specify only once instead of doing it in a loop. And, also it doesn't center now
https://www.loom.com/share/05415df45a3143c2850acf25ad97f036?sid=8eb7d7dc-73a9-412d-bde5-f7ab37a94830